editor/scene_tree_dock.cpp

```diff
@@ -1311,23 +1311,31 @@ void SceneTreeDock::perform_node_renames(Node *p_base, List<Pair<NodePath, NodeP
 				Variant p = p_base->get(propertyname);
 				if (p.get_type() == Variant::NODE_PATH) {
 
-					// Goes through all paths to check if its matching
-					for (List<Pair<NodePath, NodePath> >::Element *F = p_renames->front(); F; F = F->next()) {
+					NodePath root_path_old = p_base->get_path();
+					NodePath root_path_new = root_path_old;
 
-						NodePath root_path = p_base->get_path();
-
-						NodePath rel_path_old = root_path.rel_path_to(F->get().first);
+					for (List<Pair<NodePath, NodePath> >::Element *F = p_renames->front(); F; F = F->next()) {
+						if (root_path_old == F->get().first) {
+							root_path_new = F->get().second;
+							break;
+						}
+					}
```

Comment:
This looks like it can be moved outside the loop, no? After the second `if (!p_base)` would be a good place.
Aside from that, the code looks ok. Needs rebasing though.

(also sorry for the late review .-.)